testing/server.go

```diff
@@ -28,12 +29,25 @@ import (
 // Callers are responsible of closing the server by calling Close().
 func OpenServer(c *gc.C, session *mgo.Session, params charmstore.ServerParams) *Server {
 	db := session.DB("charmstore-testing")
+	if params.AuthUsername == "" {
+		params.AuthUsername = "charmstore-testing-user"
+	}
+	if params.AuthPassword == "" {
+		params.AuthPassword = "charmstore-testing-password"
+	}
 	handler, err := charmstore.NewServer(db, nil, "", params, charmstore.V4)
 	c.Assert(err, jc.ErrorIsNil)
+
+	srv := httptest.NewServer(handler)
 	return &Server{
-		srv:     httptest.NewServer(handler),
+		srv:     srv,
 		handler: handler,
 		params:  params,
+		client: csclient.New(csclient.Params{
```

Comment:
Doesn't this mean that the server keeps client cookies around for the lifetime of the server? I wonder if that's actually a good idea, as I can imagine that tests might
create the server as part of test suite setup and then you'd run the danger of test cross-contamination. I think I'd be inclined to keep server strictly separate from client (but provide an easy way of creating a client from the server).